cloud/services/scalesets/vmss.go

```diff
@@ -356,3 +362,17 @@ func getVMSSUpdateFromVMSS(vmss compute.VirtualMachineScaleSet) (compute.Virtual
 	err = update.UnmarshalJSON(json)
 	return update, err
 }
+
+func getSecurityProfile(vmssSpec azure.ScaleSetSpec, sku resourceskus.SKU) (*compute.SecurityProfile, error) {
+	if vmssSpec.SecurityProfile == nil {
```

Comment:
actually, I think we should default this on when possible? if a user doesn't specify it but the sku supports it, I think we should mutate the spec to default it on.

really wish we could do that in webhook rather than controller, though 🙁 wdyt @CecileRobertMichon @devigned?

Reply:
That's an interesting idea, we could follow the precedent set by `AcceleratedNetworking`, if the user explicitly sets a value then use that, otherwise set to true if the sku supports it. I don't know if we should mutate the spec though, I don't think accelerated networking does it... but maybe it should? 

This does bring up a point though that if we don't set it by default now and later decide to set it, there won't be a way to know if the value is nil because the user didn't set it or if it wasn't set before and should actually be false... so maybe we should set a default and mutate the spec, even if that default is false.

Reply:
actually, the way we do this today is broken: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/110f084337f64c49a13b01744a2e6d03dba55d13/cloud/services/networkinterfaces/networkinterfaces.go#L93

I don't think that writes back into anything which gets persisted -- it's writing back into the value from a range array, rather than writing back into the slice returned by the function. I think because the original value passed to the struct from the scope was a pointer to AcceleratedNetworking, we could technically get away with this if we ranged references instead of values? but if we try to write any non-pointer values back, that will definitely be broken since the values will have been copied inside NICScopes()

Reply:
need to test, though, also definitely out of scope for this PR

Reply:
right accelerated networking definitely does not get persisted

Reply:
> This does bring up a point though that if we don't set it by default now and later decide to set it, there won't be a way to know if the value is nil because the user didn't set it or if it wasn't set before and should actually be false... so maybe we should set a default and mutate the spec, even if that default is false

agreed, we should not allow persisting objects in final form with ambiguity (phrasing it that way since we would ideally not let them get past the webhook, but we live with that for now)

Reply:
Im ok with defaulting to on in the long run but maybe not part of this PR? As I think this might need to the testing effort?

Reply:
So in that case, should it at least get it persisted so that when we change the default later it doesn't change pre-existing machines?